lib/common/api/deprecate.js

```diff
@@ -69,23 +69,28 @@ const deprecate = {
     })
   },
 
-  promisify: (promise, cb) => {
+  promisify: (fn, numArgs) => {
     const oldName = `function with callbacks`
     const newName = `function with Promises`
     const warn = warnOnce(oldName, newName)
 
-    if (typeof cb !== 'function') return promise
-    if (process.enablePromiseAPIs) warn()
-    return promise
-      .then(res => {
-        process.nextTick(() => {
-          cb(null, res)
-        })
-      }, err => {
-        process.nextTick(() => {
-          cb(err)
+    return function (...args) {
+      const promise = fn.apply(this, args.splice(0, numArgs))
+      const cb = args[numArgs]
```

Comment:
_(bug)_ wouldn't this be `args[0]` after the previous line's splice?

And if that's correct, that `cb` and `args[numArgs]` don't point to a callback after the splice -- how did this PR pass CI, as the callback would never be invoked? Do we test `deprecate.promisify()` to confirm that the callback is called?